Feature/583 courses and faculty pages remove detail buttons

Urjikoo opened the pull request:
## Types of changes
<!--
  What types of changes does your code introduce? Put an `x` in all the boxes
  that apply:
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:
<!--
  Go over all the following points, and put an `x` in all the boxes that apply.
-->
- [X] I have run `eslint` on the code
- [ ] I have added JSDoc for all of my code (where applicable)
- [X] I have added tests to cover my changes.

## Priority:
- [X] Normal <!-- New piece of functionality -->
- [ ] High <!-- Critical bug requiring urgent review -->

## Related Issues:
<!--
  Add the issue number in below that this PR is intended to address.

  Also mention any issues that this PR is related to, and if possible,
  provide more information regarding the relationship to the issues in
  question (i.e does the PR fix the issue, is it designed to fix another bug
  that is similar to the issue etc.)
-->
Fixes #583 

<!--
  Attribution:
  PR Template adapted from: https://github.com/h5bp/html5-boilerplate/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

Urjikoo committed:
Details sections were removed from both courses and faculty pages.
Details sections were removed from courses and faculty pages as we currently don't have plans for the feature.Test was implmented removing details test as it's no longer needed.

natalynyu reviewed (changes requested):
This looks good, Urji! The `COURSE_TABLE_COLUMN` enum can be updated by removing the `DETAILS` property, and the details column can also be removed from `MANDATORY_COLUMNS`. The details object should also be removed from the `modalFields.tsx` file to remove the Details column from this Customize View modal: 
<img width="1431" alt="image" src="https://user-images.githubusercontent.com/29589689/207154498-4eb4d5a7-32a0-4897-8b8a-fdc51bf31db0.png">

Urjikoo committed:
Details section removed from customized view table on courses page.
Removed 'Details' definition from COURSE_TABLE_COLUMN, COURSE_TABLE_COLUMN_GROUP, MANDATORY_COLUMNS as a result  Details section was removed from customized view on courses page.

Urjikoo committed:
adjusted facultyScheduleTable table to resolve merge confilct.

Urjikoo committed:
facultyscheduletable file was adjusted/edited to resolve merge conflict.

Urjikoo committed:
Merge branch 'develop' into feature/583-courses-and-faculty-pages-remove-Detail-buttons

natalynyu reviewed (approved):
Looks great! Approved.
